functions/errorreporting/index.js

```diff
@@ -36,106 +36,114 @@ function reportError (err, callback) {
+ * @param {object} req Cloud Function request context.
+ * @param {object} res Cloud Function response context.
  */
 exports.helloSimpleError = function helloSimpleError (req, res) {
   try {
     if (req.method !== 'GET') {
-      var error = new Error('Only GET requests are accepted!');
+      const error = new Error('Only GET requests are accepted!');
       error.code = 405;
       throw error;
     }
     // All is good, respond to the HTTP request
-    return res.send('Hello World!');
+    res.send('Hello World!');
   } catch (err) {
     // Report the error
-    return reportError(err, function () {
+    reportError(err, () => {
       // Now respond to the HTTP request
-      return res.status(error.code || 500).send(err.message);
+      res.status(err.code || 500).send(err.message);
     });
   }
 };
-// [END helloSimpleError]
+// [END functions_errorreporting_simple]
 
-// [START helloHttpError]
+// [START functions_errorreporting_http]
 /**
  * HTTP Cloud Function.
  *
- * @param {Object} req Cloud Function request object.
- * @param {Object} res Cloud Function response object.
+ * @param {object} req Cloud Function request context.
+ * @param {object} req.body The request body.
+ * @param {string} req.body.message Message provided in the request.
+ * @param {object} res Cloud Function response context.
  */
 exports.helloHttpError = function helloHttpError (req, res) {
   try {
     if (req.method !== 'POST' && req.method !== 'GET') {
-      var error = new Error('Only POST and GET requests are accepted!');
+      const error = new Error('Only POST and GET requests are accepted!');
       error.code = 405;
       throw error;
     }
     // All is good, respond to the HTTP request
-    return res.send('Hello ' + (req.body.message || 'World') + '!');
+    res.send(`Hello ${req.body.message || 'World'}!`);
   } catch (err) {
     // Set the response status code before reporting the error
     res.status(err.code || 500);
     // Report the error
-    return reportDetailedError(err, req, res, function () {
+    reportDetailedError(err, req, res, () => {
       // Now respond to the HTTP request
-      return res.send(err.message);
+      res.send(err);
     });
   }
 };
-// [END helloHttpError]
+// [END functions_errorreporting_http]
 
-// [START helloBackgroundError]
+// [START functions_errorreporting_background]
 /**
  * Background Cloud Function.
  *
- * @param {Object} context Cloud Function context object.
- * @param {Object} data Request data, provided by a trigger.
- * @param {string} data.message Message, provided by the trigger.
+ * @param {object} event The Cloud Functions event.
+ * @param {object} event.data The event data.
+ * @param {string} event.data.message Message, provided by the trigger.
+ * @param {function} The callback function.
  */
-exports.helloBackgroundError = function helloBackgroundError (context, data) {
+exports.helloBackgroundError = function helloBackgroundError (event, callback) {
```

Comment:
The comment is supposed to be there, indicating that the user would put their own code there.